plugins/modules/vm_snapshot_attach_disk.py

```diff
@@ -0,0 +1,238 @@
+    source_snapshot_uuid = module.params["source_snapshot_uuid"]
+    source_disk_type = module.params["source_disk_type"]
+    source_disk_slot = module.params[
+        "source_disk_slot"
+    ]  # the higher the index, the newer the disk
+
+    # =============== IMPLEMENTATION ===================
+
+    vm_snapshot = VMSnapshot.get_snapshots_by_query(
+        {"uuid": source_snapshot_uuid}, rest_client
+    )[0]  # there is only 1 snapshot with the specified uuid.
+    vm_uuid = vm_snapshot["vm"]["uuid"]
+
+    # get wanted source device snapshot
+    source_device_snapshot = VMSnapshot.filter_snapshots_by_params(
+        dict(
+            source_disk_type=source_disk_type,
+            source_disk_slot=source_disk_slot,
+        ),
+        rest_client
+    )
+
+    # TODO: add "device_snapshots" to module_utils/vm_snapshot.py
+    #  add "device_snapshots" dict to the vm_snapshot from_hypercore() method
+    #  add "vm.block_devices" dict to the vm_snapshot from_hypercore() method
+    source_disk_uuid = vm_snapshot["device_snapshots"][source_disk_slot]
+
+    # TODO: implement this method!
+    #  should return a dict (NOT list) ==> I updated filter_snapshots_by_params to be able to filter by source_disk_type, slot, uuid
+    source_disk_info = vm_snapshot.get_device_snapshot(
+        dict(uuid=source_disk_uuid), rest_client
+    )[1]
+
+    # build a payload according to /rest/v1/VirDomainBlockDevice/{uuid}/clone documentation
+    payload = dict(
+        options=dict(
+            regenerateDiskID=True,
+            readOnly=source_disk_info["read_only"],
+        ),
+        snapUUID=source_snapshot_uuid,
+        template=dict(
+            virDomainUUID=vm_uuid,
+            # type=source_disk_info["type"], # --> omit for now
+            chacheMode=source_disk_info["cache_mode"],
+            slot=source_disk_info["slot"],
+            disableSnapshotting=source_disk_info["disable_snapshotting"],
+            tieringPriorityFactor=source_disk_info["tiering_priority_factor"],
+        ),
+    )
+
+    create_task_tag = rest_client.create_record(
+        endpoint="{0}/{1}/clone".format(
+            "/rest/v1/VirDomainBlockDevice", source_disk_uuid
+        ),
+        payload=payload,
+        check_mode=module.check_mode,
+    )
+
+    TaskTag.wait_task(rest_client, create_task_tag)
+    created_block_device = VMSnapshot.get_vm_disk_info(create_task_tag["createdUUID"])
+
+    # return changed, after, diff
+    return (
+        created_block_device is not None,  # if new block device was created, then this should not be None
+        created_block_device,
+        dict(
+            before={}, after=created_block_device
+        ),  # before, we ofcourse, didn't have that new block device, and after we should have it
+    )
+
+
+def run(
+    module: AnsibleModule, rest_client: RestClient
+) -> List[Optional[TypedVMSnapshotToAnsible]]:
+    filtered = VMSnapshot.filter_snapshots_by_params(module.params, rest_client)
```

Comment:
A. I didn't notice until now `attach_disk` is not yet called. Hm...

Reply:
It should be called now, and it also should work :)
Unit tests and integration tests for this module (``vm_snapshot_attach_disk``) are on the way.